src/modules/ims_ipsec_pcscf/ipsec.c

```diff
@@ -100,8 +100,15 @@ static void string_to_key(char* dst, const str key_string)
     }
 }
 
+static uint choose_nlmsg_seq (void)
```

Comment:
Any particular reason why you choose to implement this function instead of the previous time(0) call? The clock_gettime() function is not available at Mac OS, and probably need a wrapper around it as e.g. in the cdp module. If not necessary, we should probably stay with the previous way.

Reply:
Hi @henningw ,
I choose the 1 micro seconds granulation due to the commit comment:
"improve nlmsg_seq choice for concurrent multi UEs Registrations at same time"

The reason is the previous choice of time(NULL) return a time value on seconds and it can be too large in presence of multiple UE access in the same time. In high load condition it can be happens.
IPsec Security Association must have this parameter as a unique identifier, since kamailio can be configured to listen on single local ip:port for IPsec, it can be a problem, resulting in a failure on activation of SA, and then a failure on VoLTE SIP REGISTRATION.

A time based choice is a good choice, but 1 second is not enough, so I propose the choice of 1 micro second granularity that can be protect the registrations in a reasonable way.

Regarding other platforms that don't have the funtion, feel free to add a pre processor condition on the function

Reply:
Was resolved as per other discussions.